librad/src/meta/entity.rs

```diff
@@ -526,33 +483,32 @@ where
         key: &PublicKey,
         by: &Signatory,
         signature: &Signature,
-        resolver: &impl Resolver<User<EntityStatusUnknown>>,
-    ) -> Result<(), Error> {
+        resolver: &impl Resolver<User<Unknown>>,
+    ) -> Result<Entity<T, Signed>, Error>
+    where
+        ST: Clone,
+    {
         self.check_key(key, by, resolver).await?;
         if signature.verify(&self.canonical_data()?, key) {
-            Ok(())
+            Ok(self.clone().with_status::<Signed>())
         } else {
             Err(Error::SignatureVerificationFailed)
         }
     }
 
-    fn with_status<NewSt>(&self, status_marker: NewSt) -> Entity<T, NewSt>
-    where
-        NewSt: VerificationStatusMarker,
-    {
+    fn with_status<NewSt>(self) -> Entity<T, NewSt> {
         Entity::<T, NewSt> {
-            status: NewSt::status(),
-            status_marker,
-            name: self.name.clone(),
+            status_marker: PhantomData,
```

Comment:
This is what I was missing! I just did not realize that I could use `PhantomData` as a "value" and the compiler would infer the correct type.
And this was the only reason why I did not use `PhantomData ` and I essentially reimplemented it.

Reply:
Ya, it's a bit of a weird one to get your head around immediately :grin: